modules/markup/html.go

```diff
@@ -317,48 +317,52 @@ func RenderEmoji(
 	return ctx.postProcess(rawHTML)
 }
 
-var byteBodyTag = []byte("<body>")
-var byteBodyTagClosing = []byte("</body>")
-
 func (ctx *postProcessCtx) postProcess(rawHTML []byte) ([]byte, error) {
 	if ctx.procs == nil {
 		ctx.procs = defaultProcessors
 	}
 
-	// give a generous extra 50 bytes
-	res := make([]byte, 0, len(rawHTML)+50)
-	res = append(res, byteBodyTag...)
-	res = append(res, rawHTML...)
-	res = append(res, byteBodyTagClosing...)
-
 	// parse the HTML
-	nodes, err := html.ParseFragment(bytes.NewReader(res), nil)
+	nodes, err := html.ParseFragment(bytes.NewReader(rawHTML), nil)
 	if err != nil {
 		return nil, &postProcessError{"invalid HTML", err}
 	}
 
+	var newNodes = make([]*html.Node, 0, len(nodes))
 	for _, node := range nodes {
-		ctx.visitNode(node, true)
+		if node.Data == "html" {
+			node = node.FirstChild
+			for node != nil && node.Data != "body" {
+				node = node.NextSibling
+			}
+		}
+		if node.Data == "body" {
+			node = node.FirstChild
+			for node != nil {
+				newNodes = append(newNodes, node)
+				node = node.NextSibling
+			}
+		} else {
+			newNodes = append(newNodes, node)
+		}
 	}
 
 	// Create buffer in which the data will be placed again. We know that the
 	// length will be at least that of res; to spare a few alloc+copy, we
 	// reuse res, resetting its length to 0.
-	buf := bytes.NewBuffer(res[:0])
+	buf := bytes.NewBuffer([]byte{})
```

Comment:
https://golang.org/pkg/bytes/#NewBuffer

NewBuffer creates and initializes a new Buffer using buf as its initial contents. The new Buffer takes ownership of buf, and the caller should not use buf after this call. NewBuffer is intended to prepare a Buffer to read existing data. It can also be used to set the initial size of the internal buffer for writing. To do that, buf should have the desired capacity but a length of zero.

In most cases, new(Buffer) (or just declaring a Buffer variable) is sufficient to initialize a Buffer.

```suggestion
	// Create buffer in which the data will be placed again.
	buf :=  new(bytes.Buffer)
```